src/sequelize.js

```diff
@@ -193,63 +191,7 @@ export class Sequelize {
       config = {};
       options = username || {};
 
-      const urlParts = url.parse(arguments[0], true);
-
-      options.dialect = urlParts.protocol.replace(/:$/, '');
-      options.host = urlParts.hostname;
-
-      if (options.dialect === 'sqlite' && urlParts.pathname && !urlParts.pathname.startsWith('/:memory')) {
-        const storagePath = path.join(options.host, urlParts.pathname);
-        options.storage = path.resolve(options.storage || storagePath);
-      }
-
-      if (urlParts.pathname) {
-        config.database = urlParts.pathname.replace(/^\//, '');
-      }
-
-      if (urlParts.port) {
-        options.port = urlParts.port;
-      }
-
-      if (urlParts.auth) {
-        const authParts = urlParts.auth.split(':');
-
-        config.username = authParts[0];
-
-        if (authParts.length > 1) {
-          config.password = authParts.slice(1).join(':');
-        }
-      }
-
-      if (urlParts.query) {
-        // Allow host query argument to override the url host.
-        // Enables specifying domain socket hosts which cannot be specified via the typical
-        // host part of a url.
-        if (urlParts.query.host) {
-          options.host = urlParts.query.host;
-        }
-
-        if (options.dialectOptions) {
-          Object.assign(options.dialectOptions, urlParts.query);
-        } else {
-          options.dialectOptions = urlParts.query;
-          if (urlParts.query.options) {
-            try {
-              const o = JSON.parse(urlParts.query.options);
-              options.dialectOptions.options = o;
-            } catch {
-              // Nothing to do, string is not a valid JSON
-              // an thus does not need any further processing
-            }
-          }
-        }
-      }
-
-      // For postgres, we can use this helper to load certs directly from the
-      // connection string.
-      if (['postgres', 'postgresql'].includes(options.dialect)) {
-        Object.assign(options.dialectOptions, pgConnectionString.parse(arguments[0]));
-      }
+      parseConnectionString(arguments[0], options);
```

Comment:
We try to reduce how many of our methods mutate their inputs, it would be better to merge `parseConnectionString`'s output instead of giving it an object to mutate:

```suggestion
      _.merge(options, parseConnectionString(arguments[0]));
```

Reply:
Should I get rid of the options argument from parseConnectionString because that's the only place it was used?

Reply:
I took the argument out, it's superfluous but I think more importantly the function without that argument is more true to its name since now it exclusively parses the connection string you feed it.